subprojects/build-operations/src/main/java/org/gradle/trace/pid/PidCollectorJava8BuildService.java

```diff
@@ -0,0 +1,28 @@
+package org.gradle.trace.pid;
+
+import java.lang.management.ManagementFactory;
+import java.lang.management.RuntimeMXBean;
+import java.lang.reflect.InvocationTargetException;
+
+public abstract class PidCollectorJava8BuildService extends AbstractPidCollectorBuildService {
+
+    @Override
+    protected Long getPid() {
+        try {
+            // With Gradle 6.6 we can't inject `ProcessEnvironment`, so that is why we access the runtime MXBean here.
+            // On Java 9 we could use the nicer ProcessHandle.
```

Comment:
It's not "could" anymore. Maybe also point to `PidCollectorBuildService` from here?

Reply:
Good catch! I adjusted the comment.

Reply:
gradle-profiler --benchmark --profile async-profiler --project-dir ./ compileTestJava

Reply:
subprojects/build-operations/src/main/java/org/gradle/trace/pid/PidCollectorBuildService.java[](url)